src/components/EnvelopeList.vue

```diff
@@ -38,16 +61,118 @@ export default {
 		refreshing: {
 			type: Boolean,
 			required: true,
+			default: true,
 		},
 		loadingMore: {
 			type: Boolean,
 			required: true,
 		},
 	},
+	data() {
+		return {
+			shortkeys: {
+				del: ['del'],
+				flag: ['s'],
+				next: ['arrowright'],
+				prev: ['arrowleft'],
+				refresh: ['r'],
+				unseen: ['u'],
+			},
+			selection: [],
+		}
+	},
+	computed: {
+		multipleSelection() {
+			// returns true if multiple envelopes are selected
+			return this.selection.length > 0
+		},
+	},
+	methods: {
+		markSelectedSeenOrUnseen(seenFlag) {
+			this.selection.forEach(envelopeId => {
+				this.$store.dispatch('markEnvelopeSeenOrUnseen', {
+					envelope: this.envelopes[envelopeId],
+					seenFlag: seenFlag,
+				})
+			})
+			this.selection = []
+		},
+		deleteAllSelected() {
+			this.selection.forEach(envelopeId => {
+				// Navigate if the message beeing deleted is the one currently viewed
+				if (this.envelopes[envelopeId].uid == this.$route.params.messageUid) {
+					let next
+					if (envelopeId === 0) {
+						next = this.envelopes[envelopeId + 1]
+					} else {
+						next = this.envelopes[envelopeId - 1]
+					}
+
+					if (next) {
+						this.$router.push({
+							name: 'message',
+							params: {
+								accountId: this.$route.params.accountId,
+								folderId: this.$route.params.folderId,
+								messageUid: next.uid,
+							},
+						})
+					}
+				}
+				this.$store.dispatch('deleteMessage', this.envelopes[envelopeId])
+			})
+			this.selection = []
+		},
+		isEnvelopeSelected(idx) {
+			if (this.selection.length == 0) {
+				return false
+			}
+
+			return this.selection.includes(idx)
+		},
+		onEnvelopeSelected(envelope, shiftKey) {
+			const idx = this.envelopes.indexOf(envelope)
+
+			// If this is the first selected envelope, or the shift key is not pressed, simply add/remove the envelope ID to/from the selection array
+			if (!shiftKey || this.selection.length == 0) {
+				if (!this.selection.includes(idx)) {
+					this.selection.push(idx)
+				} else {
+					this.selection.splice(this.selection.indexOf(idx), 1)
+				}
+				return
+			}
+
+			// Otherwise, add (or remove) all envelopes between the last selected envelope and this one
+			const lastEnv = this.selection[this.selection.length - 1]
+			var start = lastEnv + 1
+			var last = idx
+			if (lastEnv > idx) {
+				start = idx
+				last = lastEnv - 1
+			}
+			for (var i = start; i <= last; i++) {
+				if (!this.selection.includes(i)) {
+					this.selection.push(i)
+				} else {
+					this.selection.splice(this.selection.indexOf(i), 1)
+				}
+			}
+			return
+		},
+	},
 }
 </script>
 
 <style scoped>
+.multiselect-header {
+	display: flex;
+	flex-direction: row;
+	align-items: center;
+	justify-content: center;
+	background: #0080ff;
```

Comment:
The background should be the same as for the files controls bar :)
```suggestion
	background-color: var(--color-main-background-translucent);
```

Reply:
It's just plain white?

![image](https://user-images.githubusercontent.com/8179031/76065170-62266e00-5f8b-11ea-959b-23214801653d.png)

Reply:
@StCyr exactly, until you scroll when it overlays the list translucent. :) That’s how it should be.
In reference to my comment at https://github.com/nextcloud/mail/pull/2619#issuecomment-595652200 – if we show 1 or 2 actions directly in there like "Mark 7 unread", we can use the `primary` class on the action so it appears blue and calls attention. But the background should be plain. :)